src/json-observe.ts

```diff
@@ -166,71 +179,93 @@ class JsonObserver {
     the proxyifying operation itself is being
     recorded, which in an unwanted behavior,
     that's why we disable recording through this
-    inital process;
+    initial process;
     */
 
-    this.disableCallback = true;
+    this.disableEnableCallback(false);
     var proxifiedObject = this._proxifyObjectTreeRecursively(root, "");
 
     /* OK you can record now */
-    this.disableCallback = false;
+    this.disableEnableCallback(true);
 
     return proxifiedObject;
   }
-  private equalsEvents(op1, op2): boolean {
-    if (op1.value && op2.value)
-      return op1.op === op2.code && op1.path === op2.path && op1.value === op2.value;
 
-    return op1.op === op2.code && op1.path === op2.path;
-  }
   constructor(root) {
     this.originalObject = root;
     this.cachedProxy = null;
-    this.patches = [];
     this.isRecording = false;
     this.userCallback;
-
-    this.defaultCallback = function (sender) {
-      return function (event) {
-        //don't duplicate exact same events
-        //if(sender.patches.length > 0 && this.equalsEvents(sender.patches[sender.patches.length - 1], event)) return;
-
-        if (!sender.disableCallback) {
-          if (sender.isRecording) {
-            sender.patches.push(event);
-          }
-          if (sender.userCallback) {
-            sender.userCallback(event);
-          }
+    var sender = this;
+    /* 
+    instead of setting a boolean to
+    be checked every time an operation
+    is being recording, I'm removing the 
+    whole callback when it's disabled.
+    Because we are only disabling the callback
+    once in the beginning, no need to `if` every time
+    */
+    this.disableEnableCallback = function(enable: boolean)
```

Comment:
It's not the same meaning, because it takes a boolean, doesn't technically toggles. I can suggest `switchObserver(on: boolean)`.